obs-mcp/pkg/prometheus/guardrails.go

```diff
@@ -102,27 +119,155 @@ func (g *Guardrails) IsSafeQuery(query string) (bool, error) {
 				// Rule 3: Check for expensive regex matchers on *any* label i.e blanket matchers
 				if g.DisallowBlanketRegex {
 					isRegex := m.Type == labels.MatchRegexp || m.Type == labels.MatchNotRegexp
-					if isRegex {
-						if m.Value == ".*" || m.Value == ".+" {
-							foundUnsafe = true
-							return fmt.Errorf("unsafe")
+					if isRegex && (m.Value == ".*" || m.Value == ".+") {
```

Comment:
can't we already use `ExtractBlanketRegexLabels` here and reduce this code duplicity?